src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -51,6 +85,51 @@ AppHost::~AppHost()
     _app = nullptr;
 }
 
+// Method Description:
+// - Retrieve any commandline args passed on the commandline, and pass them to
+//   the app logic for processing.
+// - If the logic determined there's an error while processing that commandline,
+//   display a message box to the user with the text of the error, and exit.
+//    * We display a message box because we're a Win32 application (not a
+//      console app), and the shell has undoubtedly returned to the foreground
+//      of the console. Text emitted here might mix unexpectedly with output
+//      from the shell process.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void AppHost::_handleCommandlineArgs()
+{
+    if (auto commandline{ GetCommandLineW() })
+    {
+        int argc = 0;
+
+        // Get the argv, and turn them into a hstring array to pass to the app.
+        LPWSTR* argv = CommandLineToArgvW(commandline, &argc);
+        if (argc > 0)
+        {
+            std::vector<winrt::hstring> args;
+            for (auto i = 0; i < argc; i++)
+            {
+                args.push_back({ argv[i] });
+            }

```

Comment:
why not call [emplace_back](https://en.cppreference.com/w/cpp/container/vector/emplace_back)